documentation/assets/Losses.md

```diff
@@ -0,0 +1,155 @@
+# Losses in SG
+
+SuperGradients provides multiple Loss function implementations for various tasks:
+
+    cross_entropy
+    mse
+    r_squared_loss
+    shelfnet_ohem_loss
+    shelfnet_se_loss
+    yolox_loss
+    yolox_fast_loss
+    ssd_loss
+    stdc_loss
+    bce_dice_loss
+    kd_loss
+    dice_ce_edge_loss
+
+All of the above, are just string aliases for the underlying torch.nn.Module classes, implementing the specified loss functions.
+
+##Basic Usage of Implemented Loss Functions in SG:
+
+When using configuration files, for example training using train_from_recipe (or similar, when the underlying train method that is being called is Trainer.train_from_config(...)):
+    
+
+In your `my_training_hyperparams.yaml` file:
+  ```yaml
+    ...
+    ...
+    
+    loss: yolox_loss
+    
+    criterion_params:
+      strides: [8, 16, 32]  # output strides of all yolo outputs
+      num_classes: 80
+  ```
+  
+   `criterion_params` dictionary will be unpacked to the underlying `yolox_loss` class constructor.
+
+
+Another usage case, is when using a direct Trainer.train(...) call:
+   - In your `my_training_script.py`
+   ```python
+        ...
+        trainer = Trainer("external_criterion_test")
+        train_dataloader = ...
+        valid_dataloader = ...
+        model = ...
+  
+        train_params = {
+            ...
+            "loss": "cross_entropy",
+            "criterion_params": {}
+            ...
+        }
+        trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+   ```
+   Note that object names in SG are not case-sensitive nor symbol-sensitive, so `"CrossEntropy` could have been passed as well.
+   Since most IDEs support auto-completion, for your convenience, you can use our object_names module:
+  ```python
+     from super_gradients.common.object_names import Losses
+  ```
+  Then simply instead of "cross_entropy", use 
+  ```python
+  Losses.CROSS_ENTROPY
+  ```
+
+## Using Your Own Loss
+SuperGradients also supports user-defined loss functions assuming they are torch.nn.Module inheritors, and that their `forward` signature is in the form:
+
+```python
+import torch.nn
+
+MyLoss(torch.nn.Module):
+...
+forward(preds, target):
+...
+
+```
+And as the argument names suggest- the first argument is the model's output, and target is the label/ground truth (argument naming is arbitrary and does not need to be specifically 'preds' or 'target').
```

Comment:
"suggest-"  -> "suggest,"

Reply:
what about the expected output format?
please add something about `component_names` and returning the loss components

Reply:
fixed and re-ordered in 2 last commits.